packages/grid/src/GridMetricCalculator.ts

```diff
@@ -1837,36 +1837,70 @@ export class GridMetricCalculator {
     return columnWidth;
   }
 
-  getTextWidth(
+  /**
+   * Calculates the width of a string using widths of individual and pairs of characters to take into account font kerning
+   * @param context The canvas rendering context
+   * @param font The font to get the width for
+   * @param text The text to calculate the width for
+   * @param maxWidth The maximum width to calculate to
+   */
+  calculateTextWidth(
     context: CanvasRenderingContext2D,
     font: string,
     text: string,
     maxWidth?: number
   ): number {
+    if (text.length === 0) return 0;
+
     if (!this.allCharWidths.has(font)) {
       this.allCharWidths.set(font, new Map());
     }
-
     const charWidths = getOrThrow(this.allCharWidths, font);
 
-    let width = 0;
+    let result = 0;
     for (let i = 0; i < text.length; i += 1) {
       const char = text[i];
+      const nextChar = i < text.length - 1 ? text[i + 1] : undefined;
```

Comment:
No need to check against length here, going past the end of the array will return `undefined` anyway.
```suggestion
      const nextChar = text[i + 1];
```